common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -549,6 +549,51 @@ public int findInSet(UTF8String match) {
     return 0;
   }
 
+  public int findInSet(UTF8String match, int collationId) {
+    if (CollationFactory.fetchCollation(collationId).supportsBinaryEquality) {
+      return this.findInSet(match);
+    }
+    if (collationId == CollationFactory.UTF8_BINARY_LCASE_COLLATION_ID) {
```

Comment:
@dbatomic do we have a general principle for this special collation? Always lower-case first and then reuse existing UTF8String functions? Will we have more collations like this in the future?

Reply:
We don't have to expect more special collations besides `UTF8_BINARY_LCASE`. Everything else will be based on ICU.

Reply:
That being said, I also don't like current direction of pushing everything into `UTF8String`.  Let me see if we can come up with some cleaner approach.

Reply:
We can solve this in the same way we did i t here: [STRING_LOCATE](https://github.com/apache/spark/pull/45791). In the next PRs we updated `CollationFactory` to be able to return "lowercase collator" so we can unify the way we deal with collated strings. If you prefer, I can update this PR with that change.

Reply:
"lowercase collator" SGTM

Reply:
@cloud-fan - Let's put on hold string expression development until we provide proper design. @uros-db, @miland-db and I will follow up on this today.